gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -399,6 +400,7 @@ public synchronized void setActive(boolean active) {
       } else { //Mark the DagManager inactive.
         log.info("Inactivating the DagManager. Shutting down all DagManager threads");
         this.scheduledExecutorPool.shutdown();
+        RootMetricContext.get().removeMatching(new MetricNameRegexFilter(ServiceMetricNames.GOBBLIN_SERVICE_PREFIX + "\\..*\\." + ServiceMetricNames.RUNNING_STATUS));
```

Comment:
let's add a comment here about background info (since a bit complex how several pieces conspire to create the issue)... something like:

the DMThread's metrics mappings follow the lifecycle of the DMThread itself and so are lost by DM deactivation-reactivation.  the `RootMetricContext` OTOH is a (persistent) singleton.  to avoid XYZ exception by the RMC preventing (re-)add of a metric already known, remove all metrics that a new DMThread thread would attempt to add (in `DagManagerThread::initialize`) whenever running post-re-enablement